gobblin-compaction/src/main/java/gobblin/compaction/dataset/DatasetsFinder.java

```diff
@@ -68,8 +68,8 @@ public DatasetsFinder(State state) {
     this.inputDir = getInputDir();
     this.destDir = getDestDir();
     this.tmpOutputDir = getTmpOutputDir();
-    this.blacklist = getBlacklist();
```

Comment:
Due to the new methods in `DatasetFilterUtils` it is possible to simplify the logic in `DatasetsFinder` and `MRCompactor`, so I put them together.